snyk/cli
# Merge internal and external (user defined) no_proxy

#### What does this PR do?
* merging internal and external defined no_proxy environment variables ensures that externally specified settings are preserved internally.
* Update snyk-docker-plugin that supports restoring system proxy settings

#### Where should the reviewer start?
internal/cliv2/cliv2.go

#### How should this be manually tested?
* Specify NO_PROXY=localhost,googleapis.com,somethingelse
* Run the clli with -d, check the new logging that should contain the user defined as well as internal no_proxy settings
2022/12/28 15:52:31.349900 legacycli:1 -    NO_PROXY = localhost,127.0.0.1,::1,googleapis.com,somethingelse

#### Any background context you want to provide?
For example when running remote docker in circleci, the external NO_PROXY configuration ensures that no traffic is being routed through any proxy.
